src/ansys/dpf/composites/_composite_model_impl.py

```diff
@@ -0,0 +1,613 @@
+            )
+
+        if measure == FailureMeasureEnum.INVERSE_RESERVE_FACTOR:
+            return max_merger.outputs.merged_fields_container()
+        else:
+            return min_merger.outputs.merged_fields_container()
+
+    @_deprecated_composite_definition_label
+    def get_sampling_point(
+        self,
+        combined_criterion: CombinedFailureCriterion,
+        element_id: int,
+        time: Optional[float] = None,
+        composite_definition_label: Optional[str] = None,
+    ) -> SamplingPointNew:
+        """Get a sampling point for an element ID and failure criteria.
+
+        Parameters
+        ----------
+        combined_criterion:
+            Combined failure criterion to evaluate.
+        element_id:
+            Element ID or label of the sampling point.
+        time:
+            Time or frequency at which to evaluate the sampling point. If ``None``,
+            the last time or frequency in the result file is used.
+        composite_definition_label:
+            Label of the composite definition, which is the
+            dictionary key in the :attr:`.ContinuousFiberCompositesFiles.composite`
+            attribute. This parameter is only required for assemblies.
+            See the note about assemblies in the description for the :class:`CompositeModel` class.
+        """
+        time_in = time
+
+        return SamplingPointNew(
+            "Sampling Point",
+            element_id,
+            combined_criterion,
+            self._material_operators,
+            self.get_mesh(),
+            self._layup_provider,
+            self._get_rst_streams_provider(),
+            self._data_sources.rst,
+            time_in,
+        )
+
+    @_deprecated_composite_definition_label
+    def get_element_info(
+        self, element_id: int, composite_definition_label: Optional[str] = None
+    ) -> Optional[ElementInfo]:
+        """Get element information for an element ID.
+
+        This method returns ``None`` if the element type is not supported.
+
+        Parameters
+        ----------
+        element_id:
+            Element ID or label.
+        composite_definition_label:
+            Label of the composite definition, which is the
+            dictionary key in the :attr:`.ContinuousFiberCompositesFiles.composite`
+            attribute. This parameter is only required for assemblies.
+            See the note about assemblies in the description for the :class:`CompositeModel` class.
+        """
+        return self._element_info_provider.get_element_info(element_id)
+
+    @_deprecated_composite_definition_label
+    def get_property_for_all_layers(
+        self,
+        layup_property: LayerProperty,
+        element_id: int,
+        composite_definition_label: Optional[str] = None,
+    ) -> Optional[NDArray[np.double]]:
+        """Get a layer property for an element ID.
+
+        Returns a numpy array with the values of the property for all the layers.
```

Comment:
```suggestion
        This method returns a numpy array with the values of the property for all the layers.
```
The longer description should be always be a complete sentence.